repository/student/student.go

```diff
@@ -0,0 +1,89 @@
+// Package student provides the data structures and repository methods
+// for managing student records.
+package student
+
+import (
+	"context"
+	"database/sql"
+	"log"
+	"time"
+)
+
+// Student represents a student record in the database.
+type Student struct {
+	ID        string
+	FirstName string
+	LastName  string
+	Grade     int32
+	CreatedAt time.Time
+}
+
+// Repository manages operations with students in the DB.
+type Repository struct {
+	db *sql.DB
+}
+
+// NewRepository creates a new Repository with the given DB connection.
+func NewRepository(db *sql.DB) *Repository {
+	return &Repository{db: db}
+}
+
+// Create inserts a new student record into the database.
+func (r *Repository) Create(ctx context.Context, s Student) (string, error) {
+	_, err := r.db.ExecContext(ctx,
+		`INSERT INTO students (id, first_name, last_name, grade, created_at)
+		 VALUES ($1, $2, $3, $4, $5)`,
+		s.ID, s.FirstName, s.LastName, s.Grade, s.CreatedAt,
+	)
+	return s.ID, err
+}
+
+// GetByID retrieves a student by their ID.
+func (r *Repository) GetByID(ctx context.Context, id string) (Student, error) {
+	var s Student
+	err := r.db.QueryRowContext(ctx,
+		`SELECT id, first_name, last_name, grade, created_at FROM students WHERE id = $1`, id).
+		Scan(&s.ID, &s.FirstName, &s.LastName, &s.Grade, &s.CreatedAt)
+	return s, err
+}
+
+// Update modifies an existing student record.
+func (r *Repository) Update(ctx context.Context, s Student) error {
+	_, err := r.db.ExecContext(ctx,
+		`UPDATE students SET first_name = $1, last_name = $2, grade = $3 WHERE id = $4`,
```

Comment:
Once again try to avoid listing all the columns explicitly, as every new column would require you to add it everywhere.
If it could not be achieved with UPDATE, try replacing it with UPSERT and pre-ensure that target row already exists (in transaction of course)